app/lib/components/update_password.svelte

```diff
@@ -0,0 +1,70 @@
+<script>
+  import app from 'app/app'
+  import PasswordInput from 'lib/components/password_input.svelte'
+  import { i18n, I18n } from 'modules/user/lib/i18n'
+
+  let showFlashCurrentPassword, showFlashNewPassword
+  let currentPassword = '', newPassword = ''
+
+  const updatePassword = async () => {
+    if (!currentPassword || currentPassword.length < 8) {
+      return flashCurrentPwdErr('wrong password')
+    }
+    if (!newPassword || newPassword.length < 8) {
+      return flashNewPwdErr('password should be 8 characters minimum')
+    }
+    if (newPassword === currentPassword) {
+      return flashNewPwdErr("that's the same password")
+    }
+    if (newPassword.length > 5000) {
+      return flashNewPwdErr('password should be 5000 characters maximum')
+    }
+    try {
+      await app.request('password:confirmation', currentPassword)
+    } catch (err) {
+      if (err.statusCode === 401) return flashCurrentPwdErr('wrong password')
+    }
+    try {
+      await app.request('password:update', currentPassword, newPassword)
+      showFlashNewPassword({
+        priority: 'success',
+        message: I18n('done')
+      })
+    } catch {
+      return flashCurrentPwdErr('something went wrong, try again later')
```

Comment:
*comment true for all occurrences of `catch {`*

if you ignore the error, we will never know what happened (the error won't be reported to the server): if you prefer to not show the error message to the user, at least log it
```suggestion
    } catch (err) {
      // Logs the error and report it
      log_.error(err)
      return flashCurrentPwdErr('something went wrong, try again later')
```

Reply:
right, its kind of a new feature as there was no catch originally.

> if you prefer

i dont really know what i prefer tho, any suggestion ?